service/awsconfig/v5/resource/loadbalancer/current.go

```diff
@@ -2,9 +2,85 @@ package loadbalancer
 
 import (
 	"context"
+
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/awsconfig/v5/key"
+)
+
+const (
+	cloudProviderClusterTagValue = "owned"
+	cloudProviderServiceTagKey   = "kubernetes.io/service-name"
 )
 
-// TODO
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	currentState := LoadBalancerState{}
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return currentState, microerror.Mask(err)
+	}
+
+	currentState, err = r.clusterLoadBalancers(customObject)
+	if err != nil {
+		return currentState, microerror.Mask(err)
```

Comment:
Here we also rely on `clusterLoadBalancers` to return the zero value which is quite implicit and not really enforced by anything. So there is a fair chance of messing up a buggy situation.

Reply:
Transparency is good I agree. But why would any value be used when non-nil error is returned? This is a bug itself.

Reply:
Yes it shouldn't be processed. Anyway I've updated to use pointers for the state.